nominal/core/workbook.py

```diff
@@ -27,6 +31,65 @@ def nominal_url(self) -> str:
         """Returns a link to the page for this Workbook in the Nominal app"""
         return f"{self._clients.app_base_url}/workbooks/{self.rid}"
 
+    def update(
+        self,
+        *,
+        title: str | None = None,
+        description: str | None = None,
+        is_draft: bool | None = None,
+        properties: Mapping[str, str] | None = None,
+        labels: Sequence[str] | None = None,
+    ) -> Self:
+        """Replace workbook metadata.
+        Updates the current instance, and returns it.
+
+        Only the metadata passed in will be replaced, the rest will remain untouched.
+
+        NOTE: This replaces the metadata rather than appending it. To append to labels or properties, merge them before
+        calling this method. E.g.:
+
+            new_labels = ["new-label-a", "new-label-b"]
+            for old_label in workbook.labels:
+                new_labels.append(old_label)
+            workbook = workbook.update(labels=new_labels)
+        """
+        # NOTE: not saving updated metadata response, as we deserialize from a notebook rather than
+        #       from metadata
+        self._clients.notebook.update_metadata(
+            self._clients.auth_header,
+            scout_notebook_api.UpdateNotebookMetadataRequest(
+                title=title,
+                description=description,
+                is_draft=is_draft,
+                labels=None if labels is None else [*labels],
+                properties=None if properties is None else {**properties},
+            ),
+            self.rid,
+        )
+        notebook = self.__class__._from_conjure(
+            self._clients, self._clients.notebook.get(self._clients.auth_header, self.rid)
+        )
+        update_dataclass(self, notebook, fields=self.__dataclass_fields__)
+        return self
+
+    def get_refnames(self) -> Sequence[str]:
+        """Get the list of refnames used within the workbook."""
+        return self._clients.notebook.get_used_ref_names(self._clients.auth_header, self.rid)
+
+    def update_refnames(self, refname_map: Mapping[str, str]) -> None:
+        """Updates refnames using a provided map of original refnames to the new refnames to replace them."""
+        self._clients.notebook.update_ref_names(
+            self._clients.auth_header, scout_notebook_api.UpdateRefNameRequest({**refname_map}), self.rid
+        )
+
+    def lock(self) -> None:
+        """Locks the workbook, preventing changes from being made to it."""
+        self._clients.notebook.lock(self._clients.auth_header, self.rid)
+
+    def unlock(self) -> None:
+        """Unlocks the workbook, allowing changes to be made to it."""
+        self._clients.notebook.unlock(self._clients.auth_header, self.rid)
+
```

Comment:
are these ops idempotent?

Reply:
I expect that users will generally want to write code like:

```py
workbook = get_workbook()
workbook.unlock()
workbook.update_refnames()
workbook.lock()
```

which is fine, but just want to make sure that the first `.unlock()` isn't going to conditionally throw